src/savegame_json.cpp

```diff
@@ -2948,6 +2948,51 @@ void mission::serialize( JsonOut &json ) const
     json.end_object();
 }
 
+void npc_work_log_entry::serialize( JsonOut &json ) const
+{
+    json.start_object();
+    json.member( "entry_npc_name", entry_npc_name );
+    json.member( "entry_timestamp", entry_timestamp );
+    json.member( "entry_npc_id", entry_npc_id );
+    json.member( "entry_type", static_cast<int>( entry_type ) );
```

Comment:
Saving the underlying enum value is kind of fragile. Better define `io::enum_to_string<log_entry_type>` and `enum_traits<log_entry_type>::last` and use the enum conversion utility, to save this enum as string id.

Reply:
Can you give me an example of something else doing that? I cant seem to find how to use the enum conversion.

Reply:
These two commits uses the enum conversion:

https://github.com/CleverRaven/Cataclysm-DDA/pull/37299/files/6d78d2b9ca32f745089114e78de579ad7e32e8cb..8ea605932ea710b782c108877857f91ccc5d2a15

Most of it is inside the first commit, but serialization is in the second one.

Reply:
im afraid I cant get that to work, been trying for a while, I dont understand it and im confronted with hundreds of weird linker errors saying that its previously defined, when adding the enum_to_string() and stuff .

Its hundreds more lines of code for the sake of not static casting an int to/from an enum like other things in savegame_json.cpp does?  if you really think its necessary, I could push my broken commit to see if you can see what is wrong with it?

Reply:
It's okay if you save it as an int, it just makes future maintainance and modification harder: the specific integer values stuck as long as we want to maintain back-compatibility. If you decide to settle on enum conversion, push the problematic commit and I'll check it for you.

Reply:
pushed, thanks!

Reply:
pushed thanks!